client/src/js/components/bhReportSource.js

```diff
@@ -0,0 +1,48 @@
+angular.module('bhima.components')
+.component('bhReportSource', {
+  bindings : {
+    onSelectCallback : '&?',
+    label : '@?',
+    name : '@?',
+
+    // allows the controller to set a default value
+    value : '<'
+  },
+  template :
+    '<div class="form-group">' +
+      '<label class="control-label" translate>{{$ctrl.label}}</span></label>' +
+      '<select ng-change="$ctrl.onSelect($ctrl.value)" class="form-control" ng-model="$ctrl.value" ng-options="source.id as (source.label | translate) for source in $ctrl.sources" required>' +
+        '<option value="" disabled translate>FORM.SELECT.SOURCE</option>' +
+      '</select>' +
+      '{{$ctrl.invalue}}' +
+    '</div>',
+  controller : bhReportSourceController
+});
+
+bhReportSourceController.$inject = [];
+
+function bhReportSourceController() {
+  var ctrl = this;
+  var DEFAULT_SOURCE = 1;
+
+  ctrl.label = ctrl.label || 'FORM.SELECT.SOURCE';
```

Comment:
All of these bindings must go in `ctrl.$onInit` for the controller to properly detect them in all cases.  [See this article for more information](https://toddmotto.com/angular-1-5-lifecycle-hooks#oninit) as well as the [angular documentation](https://docs.angularjs.org/guide/component)